src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
While the logic could live somewhere it would not change the end result here; this object has been disposed, the fact that it happens to call transmitter.Flush inside it's flush method is not relevant.  What is relevant is that this object has been disposed and calling anything on it once it has been disposed of should throw the ObjectDisposedException.
Even if we added the check to the transmitter, which users can't access anyway, it would still get thrown when this method is called after this object has been disposed of because this channel disposes of its private transmitter instance when the channel is disposed of.